spec/system/visit_users_spec.rb

```diff
@@ -118,14 +140,53 @@
   end
 
   context 'IDP User' do
-    before(:each) do
-      user = FactoryBot.create(:idp_user_manager_user, team: team_idp.team.id)
+    let(:team_idp) { create(:team, team_type: 'idp') }
+    let(:create_teams) { team_idp}
+  
+    let(:idp_team_member_1) { create(:idp_user_manager_user, email: 't1@t.com', team: team_idp.id) }
+    let(:idp_team_member_2) { create(:idp_user_manager_user, email: 't2@t.com', team: team_idp.id) }
+    let(:idp_team_member_3) { create(:idp_user_manager_user, email: 't3@t.com', team: team_idp.id) }
+    let(:idp_team_member_4) { create(:idp_user_manager_user, email: 't4@t.com', team: team_idp.id) }
+  
+    let(:cognito_users) {
+    {users: [
+        {username: "111",
+          user_status: 'CONFIRMED',
+          attributes: [{name: "given_name", value: idp_team_member_1.first_name},
+                      {name: "family_name", value: idp_team_member_1.last_name},
+                      {name: "email", value: idp_team_member_1.email},
+                      {name: "custom:roles", value: "usermgr"}
+          ]},
+        {username: "222",
+          user_status: 'FORCE_CHANGE_PASSWORD',
+           attributes: [{name: "given_name", value: idp_team_member_2.first_name},
+                      {name: "family_name", value: idp_team_member_2.last_name},
+                      {name: "email", value: idp_team_member_2.email},
+                      {name: "custom:roles", value: "usermgr"}
+          ]},
+        {username: "333",
+          user_status: 'COMPROMISED',
+          attributes: [{name: "given_name", value: idp_team_member_3.first_name},
+                      {name: "family_name", value: idp_team_member_3.last_name},
+                      {name: "email", value: idp_team_member_3.email},
+                      {name: "custom:roles", value: "usermgr"}
+          ]},
+          {username: "444",
+          user_status: 'RESET_REQUIRED',
+          attributes: [{name: "given_name", value: idp_team_member_4.first_name},
+                      {name: "family_name", value: idp_team_member_4.last_name},
+                      {name: "email", value: idp_team_member_4.email},
+                      {name: "custom:roles", value: "usermgr"}
+          ]}
+          ]}
+    }
+
+    it 'shows team members' do
+      user = FactoryBot.create(:idp_user_manager_user, team: team_idp.id)
       login_as(user, scope: :user)
       stub_cognito_response(method: :list_users_in_group, payload: cognito_users)
-    end
 
-    it 'shows team members' do
-      visit users_path
+      visit users_path       
       expect(page).to have_content t('users.title_for_team')+' '+team_idp.name
       cognito_users[:users].each do |user|
         if visible_statuses.include?(user[:user_status])
```

Comment:
I think conditionals in tests are a bad idea, it is often better to duplicate the test and be explicit on different behaviour based on the conditional branch you can change the test title appropriately

Reply:
I believe the reason Jakub did this was because these system tests are very expensive and we wanted to keep them to a minimal as possible, separating all of these out would add several test cases for each context and would slow down the pre-commit script. Im happy to separate them out if you think its worth it but thats the reason I believe it was done

Reply:
We can leave this out of this PR,  i still think a more readable spec is better and it depends on how slow this gets if we make it more readable. I think this is outside scope of this PR and could be a BAU task